go_modules/lib/dependabot/go_modules/file_updater/go_mod_updater.rb

```diff
@@ -154,13 +154,15 @@ def run_go_vendor
         end
 
         def run_go_get(dependencies = [])
+          # `go get` will fail if there are no go files in the directory.
+          # For example, if a `//go:build` tag excludes all files when run
+          # on a particular architecture. However, dropping a go file with
+          # a `package ...` line in it will always make `go get` succeed...
+          # Even when the package name doesn't match the rest of the files
```

Comment:
Interesting! I wonder if this behavior could break on us in the future, because it does seem a bit counter-intuitive, but I guess we can cross that bridge when we get there